syntax = "proto3";

enum Status {
        DEFAULT = 0;
        REGISTERED = 1;
        PII_REGISTERED = 2;
        SKILL_REGISTERED = 3;
        CONFIRMED = 4;
        EXPIRED = 5;
        ACK_CONFIRMED = 6;
        INVALID = 7;
        INVALID_ACK = 8;
        RECOVERY_REQ = 9;
}
message AttributeData{
        bytes value = 1;
        string signer_pub_key = 2;
        string sign_timestamp = 3;
        string valid_till = 4;
        string enc_code = 5;
}
message AttributeDataType {
    Status status = 1;
    bytes attribute_data_enc = 2; //hash of the real value + code
    string certificate = 3;  //sign on the hash of attribute_data_enc
    int32 credibility_strength = 5; //only from peer_verification, out of 100
    repeated bytes validator_list = 6;

}

message CourseDetails{

    string enc_code = 1;     //random number
    string course_ID = 2;
    string course_provider_name = 3;
    string course_name = 4;
    string course_description = 5;
    string course_start_date = 6;
    string course_finish_date = 7;
    string content_creator_list = 8;
    string score = 9;
    string course_cert_hash = 10;

}

message CourseAttributeDataType {

    bytes course_details = 1;
    string attestor_pub_key = 2;
    string sign_timestamp = 3;
    string valid_till = 4;
    // string learner_name = 5;
    string skill_point = 6;

}